daffodil-cli/src/test/scala/org/apache/daffodil/cliTest/TestCLIParsing.scala

```diff
@@ -703,4 +703,12 @@ class TestCLIparsing {
       cli.expectErr("Unable to load configuration")
     } (ExitCode.ConfigError)
   }
+
+  @Test def test_Layer_Execution_Error(): Unit = {
+    val schema = path("daffodil-cli/src/test/resources/org/apache/daffodil/layers/bad.dfdl.xsd")
+    runCLI(args"parse -s $schema") { cli =>
+      cli.sendLine("0", inputDone = true)
+      cli.expectErr("Unexpected exception in layer transformer 'bad'")
+    } (ExitCode.LayerExecutionError)
+  }
```

Comment:
It might be nice (but only a nice to have, I don't want to impose additional work if it seems unnecessary) if the bad layer code was capable of performing both a bad transformation and a good transformation and we had both bad and good layer execution tests, just to verify that the layer code is working correctly except when it throws an exception on purpose.  The trigger could be simply to send"0" for the exception or "1" for the successful transformation and the successful transformation should return some verifiable output too.

Reply:
P.S. A way to throw the exception might be to divide 1 by the input (integer division by 1 returns 1, integer division by 0 throws java.lang.ArithmeticException automatically).